parsers/CA_PE.py

```diff
@@ -0,0 +1,151 @@
+# The arrow library is used to handle datetimes consistently with other parsers
+import arrow
+
+import datetime
+
+# The request library is used to fetch content through HTTP
+import requests
+
+
+timezone = 'Canada/Atlantic'
+
+
+def _get_pei_info(requests_obj):
+    """
+    Gets current electricity flows in and out of New Brunswick.
+
+    There is no reported data timestamp in the page. The page returns
+    current time and says "Times at which values are sampled may vary by
+    as much as 5 minutes."
+    """
+
+    url = 'http://www.gov.pe.ca/energy/js/chart-values.php'
+    response = requests_obj.get(url)
+
+    raw_data = response.json()[0]
+
+    # meaning of keys per https://ruk.ca/content/open-data-pei-electricity
+    data = {
+        'pei_load': raw_data['data1'],
+        'pei_wind_gen': raw_data['data2'],
+        'pei_fossil_gen': raw_data['data3'],  # "Fossil Fueled Generation"
+        'pei_wind_used': raw_data['data4'],
+        'pei_wind_exported': raw_data['data5'],
+        'utc_datetime': arrow.get(raw_data['updateDate']).datetime
+    }
+
+    return data
+
+
+def fetch_production(country_code='CA-PE', session=None):
+    """Requests the last known production mix (in MW) of a given country
+
+    Arguments:
+    country_code       -- ignored here, only information for CA-PE is returned
+    session (optional) -- request session passed in order to re-use an existing session
+
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+
+    requests_obj = session or requests.session()
+    raw_info = _get_pei_info(requests_obj)
+
+    data = {
+        'datetime': raw_info['utc_datetime'],
+        'countryCode': country_code,
+        'production': {
+            'wind': raw_info['pei_wind_gen'],
+
+            # These are oil-fueled ("heavy fuel oil" and "diesel") generators
+            # used as peakers and back-up
+            'oil': raw_info['pei_fossil_gen']
```

Comment:
All production modes that you know for sure don't apply can be set to 0.
This will enable CO2 calculations (if too many fields are missing, we don't calculate it)